### PR TITLE
Bump to PHP 7

### PR DESCRIPTION
**Pull request by TomasVotruba**
## Why?

- majority of PHP [world bumping to PHP 7.1](https://gophp71.org/)
- upcoming [PHP 7.2](https://github.com/php-ai/php-ml/pull/147)
- we could add PHP 7.0 typehints via coding standards
- [37 % people](https://seld.be/notes/php-versions-stats-2017-2-edition) are on PHP 7.1 and growing
- PHP 5.6 and bellow will [be still supported](https://seld.be/images/update-reqs.png)
- Laravel 5.6 will require [PHP 7.1](https://github.com/laravel/framework/pull/21995)
- it's more strict and thus tested and reliable code, more pleasant to maintain and extend

### Help Needed

There are 5 issues of missed mock types, that were silently passing before strict typehint.
I tried to spend dozens of minutes over that, but I don't understand their nested system.

Any help is really appreciated :+1:

### CODE REVIEW

**Comment by tymondesigns** [comment]
Wow, thanks for putting in the effort for this!

It's been my plan to do this for `v1.1` so this will certainly come in useful.
There are also some other changes that I will be making for that version, such as deprecating the `JWTAuth` class.

I am going to add this to the 1.1 milestone for now

Thanks again

**Comment by TomasVotruba** [comment]
It was messy with those mocks (I prefer anonymous classes or better DI), but glad you find this useful.

Let me know when this is ready. I have Rector with PHPUnit and PHPStan PRs in my pocket ;)

**Comment by jerguslejko** [comment]
@tymondesigns I would really like to see `v1.0.0` to require `php7.0` as a minimal version.

**Comment by tymondesigns** [comment]
@jerguslejko `v1.1.0` will be the version that implements this

File: README.md [hunk 1]
```diff
@@ -22,11 +22,29 @@ Documentation for `1.0.0` is coming soon, but there is an unfinished guide [here
 
 If you want to easily add secure authentication to Laravel apps, feel free to check out Auth0's SDK and free plan at [auth0.com/overview](https://auth0.com/overview?utm_source=GHsponsor&utm_medium=GHsponsor&utm_campaign=jwt-auth&utm_content=auth)
 
-------------------------------------
+## License
 
-<a target='_blank' rel='nofollow' href='https://app.codesponsor.io/link/5S8eNjgwCUjgucKT89buNKez/tymondesigns/jwt-auth'>
-  <img alt='Sponsor' width='888' height='68' src='https://app.codesponsor.io/embed/5S8eNjgwCUjgucKT89buNKez/tymondesigns/jwt-auth.svg' />
-</a>
+The MIT License (MIT)
```
Comment (TomasVotruba) [inline README.md]: There is no need to duplicate already existing file: https://github.com/tymondesigns/jwt-auth/blob/develop/LICENSE

This can be removed

**Comment by TomasVotruba** [comment]
Not relevant anymore. I'm closing to clean my opened PRs.

Feel free to cherry pick anything you need.

**Comment by tymondesigns** [comment]
@TomasVotruba No problem, thanks again for doing this :+1:

**Pull request closed without merging**